src/logics/ArithLogic.cc

```diff
@@ -585,34 +586,47 @@ PTRef ArithLogic::mkPlus(vec<PTRef> && args)
     if (args.size() == 1) {
         return args[0];
     }
+    assert(args.size() != 0);
     // This code takes polynomials (+ (* v c1) (* v c2)) and converts them to the form (* v c3) where c3 = c1+c2
-    VecMap<PTRef,PTRef,PTRefHash> s2t;
-    vec<PTRef> keys;
+    Map<PTRef,uint32_t,PTRefHash> varIndices;
+    struct Entry {
+        PTRef var;
+        std::variant<PTRef,FastRational> coeff;
+    };
+    std::vector<Entry> simplified;
+    simplified.reserve(args.size());
+
     for (PTRef arg : args) {
         auto [v,c] = splitTermToVarAndConst(arg);
         assert(c != PTRef_Undef);
         assert(isConstant(c));
-        if (!s2t.has(v)) {
-            s2t.insert(v, {c});
-            keys.push(v);
-        } else
-            s2t[v].push(c);
+        if (not varIndices.has(v)) {
+            varIndices.insert(v, simplified.size());
+            simplified.push_back({.var = v, .coeff = c});
+        } else {
+            auto index = varIndices[v];
+            auto & entry = simplified[index];
+            if (std::holds_alternative<PTRef>(entry.coeff)) {
+                entry.coeff = this->getNumConst(std::get<PTRef>(entry.coeff));
```

Comment:
Do we really need this if?
Maybe just put `.coeff = this->getNumConst(c)` in line 605, if we really need this func call?

Reply:
What is the reason to call getNumConst btw?

Reply:
The point for this code is to avoid the conversion between `PTRef` and `FastRational` if possible. The conversion is only needed when there are multiple arguments of the sum with the same variable. Then we need to sum up all coefficients for the variable and for that we need coefficient terms converted to the rational numbers.